packages/flutter_tools/lib/src/base/terminal.dart

```diff
@@ -155,8 +155,8 @@ class AnsiTerminal {
   String clearScreen() => supportsColor ? clear : '\n\n';
 
   set singleCharMode(bool value) {
-    final Stream<List<int>> stdin = io.stdin;
-    if (stdin is io.Stdin && stdin.hasTerminal) {
+    if (io.stdio.hasTerminal) {
```

Comment:
`io.stdinHasTerminal` ?
Also, you could flip the sense of the test and return early then unindent.

Reply:
Done